src/routers/datatypeRouters/edges/go_terms_proteins.ts

```diff
@@ -0,0 +1,93 @@
+import { z } from 'zod'
+import { publicProcedure } from '../../../trpc'
+import { loadSchemaConfig } from '../../genericRouters/genericRouters'
+import { RouterEdges } from '../../genericRouters/routerEdges'
+import { paramsFormatType } from '../_helpers'
+import { RouterFilterBy } from '../../genericRouters/routerFilterBy'
+import { descriptions } from '../descriptions'
+
+const schema = loadSchemaConfig()
+
+const geneSchema = schema.gene
+const proteinSchema = schema.protein
+
+const goTermsProteinsSchema = schema.gaf
+const genesProteinsGoRouter = new RouterEdges(goTermsProteinsSchema)
+
+const goTermQueryFormat = z.object({
+  go_term_id: z.string(),
+  page: z.number().default(0)
+})
+
+const queryFormat = z.object({
+  query: z.string(),
+  page: z.number().default(0)
+})
+
+async function geneIds (id: string): Promise<any[]> {
+  const input: paramsFormatType = {}
+  input.gene_name = id
+  input.gene_id = id
+  input.hgnc = `HGNC:${id}`
+  input.alias = id
+  input._key = id
+
+  return await (new RouterFilterBy(geneSchema)).getObjectIDs(input, '', false)
+}
+
+async function proteinIds (id: string): Promise<any[]> {
+  const input: paramsFormatType = {}
+  input.name = id
+  input.dbxrefs = id
+  input._key = id
+
+  return await (new RouterFilterBy(proteinSchema)).getObjectIDs(input, '', false)
+}
+
+async function goTermsSearch (input: paramsFormatType): Promise<any[]> {
+  const query = input.query as string
+  const page = input.page as number
+
+  let response
+
+  // assuming an ID will match either a gene or a protein
+  const genes = await geneIds(query)
+  if (genes.length !== 0) {
+    response = await genesProteinsGoRouter.getSourceAndEdgeSet(genes[0], page)
```

Comment:
If I understand it correctly, function `getSourceAndEdgeSet` is to get source go term for given target. The target here is protein. I am not sure how can you search gene in protein collection?

Reply:
we still don't have the data for GO terms to genes, we are just leaving the support here

Reply:
I think maybe for given id for go term search, if it is protein id, get the go term using edge collection go_terms_proteins. If it is gene id, find corresponding protein id for that gene id first, then get the go term.

Reply:
I will adapt the code and do the gene -> proteins traversal and then fetch the go terms